index.js

```diff
@@ -102,6 +106,57 @@ function fastifyPostgres (fastify, options, next) {
     }
   }
 
+  if (!fastify.hasRequestDecorator('pg')) {
+    fastify.decorateRequest('pg', null)
+  }
+
+  fastify.addHook('onRoute', routeOptions => {
+    const useTransaction = routeOptions.pg && routeOptions.pg.transact
+
+    if (useTransaction) {
+      const preHandler = async (req, reply) => {
+        const client = await pool.connect()
+
+        if (name) {
+          if (client[name]) {
+            return next(new Error(`pg client '${name}' is a reserved keyword`))
+          } else if (fastify.pg[name]) {
+            return next(new Error(`request client '${name}' has already been registered`))
```

Comment:
the error handling here should be done differently from how it's done in the plugin root. calling next with an Error was fine there, it is not here. Simply throw the error instead. Same at line 130.